src/out-json.c

```diff
@@ -21,7 +21,8 @@ static void
 json_out_close(struct Output *out, FILE *fp)
 {
     UNUSEDPARM(out);
-    fprintf(fp, "]\n"); // enclose the atomic {}'s into an []
+    fseek(fp, -2, SEEK_END); // go back before last ","
```

Comment:
FWIW, fp can be stdout.

Reply:
Yeah, fseek might not be the best way to fix this... I'm not sure what else to do though.

It's just not really practical to have an invalid json output, as it breaks all the other tools that work with json. The previous version also generated invalid json files, but at least it was valid "line-by-line" json.